src/main/java/com/zendesk/maxwell/bootstrap/AsynchronousBootstrapper.java

```diff
@@ -0,0 +1,142 @@
+package com.zendesk.maxwell.bootstrap;
+
+import com.google.code.or.OpenReplicator;
+import com.zendesk.maxwell.*;
+import com.zendesk.maxwell.producer.AbstractProducer;
+import com.zendesk.maxwell.schema.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class AsynchronousBootstrapper extends AbstractBootstrapper {
+
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellReplicator.class);
+
+	private Thread thread = null;
+	private Queue<RowMap > queue = new LinkedList<>();
+	private RowMap bootstrappedRow = null;
+	private RowMapBufferByTable skippedRows = null;
+	private SynchronousBootstrapper synchronousBootstrapper = getSynchronousBootstrapper();
+
+	public AsynchronousBootstrapper( MaxwellContext context ) throws IOException {
+		super(context);
+		skippedRows = new RowMapBufferByTable();
+	}
+
+	@Override
+	public boolean isStartBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isStartBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isCompleteBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isCompleteBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isBootstrapRow(row);
+	}
+
+	protected SynchronousBootstrapper getSynchronousBootstrapper( ) {
+		return new SynchronousBootstrapper(context);
+	}
+
+	@Override
+	public boolean shouldSkip(RowMap row) throws SQLException, IOException {
+		// The main replication thread skips rows of the currently bootstrapped
+		// table and the tables that are queued for bootstrap. The bootstrap thread replays them at
+		// the end of the bootstrap. If maxwell is stopped these skipped rows will be lost;
+		// however, at next startup resume() restarts the bootstrap process from the
+		// beginning which restores the consistency of the replication stream.
+		if ( bootstrappedRow != null && haveSameTable(row, bootstrappedRow) ) {
+			skippedRows.add(row);
+			return true;
+		}
+		for ( RowMap queuedRow : queue ) {
+			if ( haveSameTable(row, queuedRow) ) {
+				skippedRows.add(row);
+				return true;
+			}
+		}
```

Comment:
ok, I missed that we queue up a bunch of bootstraps to do and then skip these too.